aries_cloudagent/core/profile.py

```diff
@@ -115,6 +116,18 @@ def inject_or(
     async def close(self):
         """Close the profile instance."""
 
+    def finalizer(self) -> Optional[finalize]:
```

Comment:
I think the finalizer can just be set up in the profile's __init__ method when required. We don't have to keep a reference to it, and requiring the caller to initialize it seems error prone.